.github/workflows/finishReleaseCycle.yml

```diff
@@ -0,0 +1,107 @@
+name: Prepare production deploy
+
+on:
+  issues:
+    types: [closed]
+
+jobs:
+  # This job is executed when a StagingDeployCash is closed.
+  # It updates the production branch to trigger the production deploy,
+  # and creates a new StagingDeployCash for the next release cycle.
+  updateProduction:
+    runs-on: ubuntu-latest
+
+    # Note: Anyone with Triage access to the Expensify.cash repo can trigger a production deploy
+    if: contains(github.event.issue.labels.*.name, 'StagingDeployCash')
+    steps:
+      - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.OS_BOTIFY_TOKEN }}
+
+      - name: Set up git config
+        run: git config user.name OSBotify
+
+      - name: Checkout staging branch
+        run: git checkout staging
```

Comment:
I think you can do this in `actions/checkout`

https://github.com/actions/checkout#Checkout-a-different-branch

Reply:
Do you know if you can provide a `ref` and the `fetch-depth: 0` to checkout a particular branch but also fetch the full branch history?

Reply:
I don't see anywhere saying you explicitly can't, so I would assume you can